x/contracts/simulator/state/manager.go

```diff
@@ -5,43 +5,44 @@ package state
 
 import (
 	"context"
-	"crypto/sha256"
 	"encoding/binary"
 	"errors"
 
 	"github.com/ava-labs/avalanchego/database"
-	"github.com/ava-labs/avalanchego/ids"
 
 	"github.com/ava-labs/hypersdk/codec"
-	"github.com/ava-labs/hypersdk/crypto/ed25519"
 	"github.com/ava-labs/hypersdk/state"
 	"github.com/ava-labs/hypersdk/x/contracts/runtime"
 )
 
 var _ runtime.StateManager = &ContractStateManager{}
 
 var (
-	ErrUnknownAccount = errors.New("unknown account")
-	balanceKeyBytes   = []byte("balance")
-	contractKeyBytes  = []byte("contract")
+	balanceKeyBytes       = []byte("balance")
+	contractManagerPrefix = []byte("contract")
 )
 
 const (
-	contractPrefix = 0x0
-
-	accountPrefix      = 0x1
-	accountDataPrefix  = 0x0
-	accountStatePrefix = 0x1
-
-	addressStoragePrefix = 0x3
+	BalanceManagerPrefix = 0x1
+	BalanceDataPrefix    = 0x0
 )
 
 type ContractStateManager struct {
-	db state.Mutable
+	db            state.Mutable
+	contractState *runtime.ContractStateManager
 }
 
 func NewContractStateManager(db state.Mutable) *ContractStateManager {
-	return &ContractStateManager{db}
+	contractManager := runtime.NewContractStateManager(
+		runtime.NewPrefixStateMutable(
```

Comment:
just have `NewContractStateManager` take a byte prefix and it can create the prefixed state.